### PR TITLE
Add proxy support for jenkins servers without direct access to the internet

### PR DESCRIPTION
**Pull request by ccloes-intuit**
We have our jenkins systems behind proxy servers internally which caused an issue for us.  I am not terribly familiar with the jenkins plugin process, but was able to compile this and get it to work.  

If this is useful, feel free to use it as necessary.  If nothing else, please add "proxy support" to your list of things to add.

### CODE REVIEW

File: src/main/resources/com/amazonaws/codedeploy/AWSCodeDeployPublisher/global.jelly [hunk 1]
```diff
@@ -12,5 +12,11 @@
       <f:entry title="AWS Secret Key" name="codedeploy" help="/plugin/codedeploy/help-awsSecretKey.html">
         <f:password field="awsSecretKey" />
       </f:entry>
+	  <f:entry title="Proxy Host" name="codedeploy" help="/plugin/codedeploy/help-proxyHost.html">
+        <f:textbox default="" field="proxyHost" />
+      </f:entry>
+      <f:entry title="AWS Secret Key" name="codedeploy" help="/plugin/codedeploy/help-proxyPort.html">
```
Comment (suryanarayanan) [inline src/main/resources/com/amazonaws/codedeploy/AWSCodeDeployPublisher/global.jelly]: Minor: You might want this to be called Proxy Port.

**Comment by suryanarayanan** [comment]
Hi,
We'll review this internally. Meanwhile I just wanted to confirm if you are okay with releasing it under Apache 2.0 license.
Thanks.

**Comment by ccloes-intuit** [comment]
Yes.  We are of course behind releasing it under the Apache 2.0 license.  

I will get the type mentioned above fixed as well.  Thanks!!!

**Comment by Jmcfar** [comment]
Thanks again for the pull request!

**Pull request merged**